src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -6854,5 +6854,40 @@ public R M4()
 ").VerifyDiagnostics(
                 );
         }
+
+        [Fact]
+        [WorkItem(39960, "https://github.com/dotnet/roslyn/issues/39960")]
+        public void MissingExceptionType()
+        {
+            var comp = CreateCompilation(@"
+class C
+{
+    void M(bool b, dynamic d)
+    {
+        _ = b
+            ? throw new System.NullReferenceException()
+            : throw null;
+        L();
+        throw null;
+        void L() => throw d;
+    }
+}
+");
+            comp.MakeTypeMissing(WellKnownType.System_Exception);
+            comp.VerifyDiagnostics(
```

Comment:
>comp.VerifyDiagnostics( [](start = 12, length = 23)

Please test also with 7.0 language version.

Reply:
Sure, but that will be testing none of the code that changed in this PR.

---
In reply to: [354033690](https://github.com/dotnet/roslyn/pull/40145#discussion_r354033690) [](ancestors = 354033690)